cdk/database/__init__.py

```diff
@@ -62,3 +67,27 @@ def dynamodb_single_table(self):
                                         partition_key=aws_dynamodb.Attribute(
                                             name='PK',
                                             type=aws_dynamodb.AttributeType.STRING))
+
+    def dynamodb_visits_table(self):
+        self.visits_table = aws_dynamodb.Table(self,
+                                               self.visits_id,
+                                               point_in_time_recovery=True,
+                                               removal_policy=core.RemovalPolicy.RETAIN,
+                                               sort_key=aws_dynamodb.Attribute(
+                                                   name='username',
```

Comment:
Personally, I would like to see our username be the pk and visit time be the sk. Think of a PK as the key to a hash map. In order to query the value for that key, you need to know the exact key. In other words, if we want to look up a specific visit to understand how the space was used, we would need to know the exact timestamp of entry.

Sort keys on the other hand are optional second attributes that we can use to filter on. https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.html

The value in having username as PK and timestamp as SK is that it enables us to look up all visits by a given user, and we can filter on things like, all visits by a user between certain dates, or in the year 2021.

Ultimately, we will use athena and quicksight to display timeseries data but it isnt necessary to store the data in DDB as a time series if that makes sense. We want to store the data in such a way that simplifies our API interactions with the database